src/dialects/abstract/data-types.ts

```diff
@@ -0,0 +1,2331 @@
+    };
+
+    if (this.options.precision != null && (this.options.precision < 0 || !Number.isInteger(this.options.precision))) {
+      throw new TypeError('Option "precision" must be a positive integer');
+    }
+  }
+
+  toSql() {
+    if (this.options.precision != null) {
+      return `DATETIME(${this.options.precision})`;
+    }
+
+    return 'DATETIME';
+  }
+
+  validate(value: any) {
+    if (!Validator.isDate(String(value))) {
+      ValidationErrorItem.throwDataTypeValidationError(
+        util.format('%O is not a valid date', value),
+      );
+    }
+  }
+
+  sanitize(value: unknown): unknown {
+    if (value instanceof Date || dayjs.isDayjs(value) || isMoment(value)) {
+      return value;
+    }
+
+    if (typeof value === 'string' || typeof value === 'number') {
+      return new Date(value);
+    }
+
+    throw new TypeError(`${util.inspect(value)} cannot be converted to a Date object, and is not a DayJS nor Moment object`);
+  }
+
+  parseDatabaseValue(value: unknown): unknown {
+    return this.sanitize(value);
+  }
+
+  areValuesEqual(
+    value: AcceptedDate,
+    originalValue: AcceptedDate,
+  ): boolean {
+    if (
+      originalValue
+      && Boolean(value)
+      && (value === originalValue
+        || (value instanceof Date
+          && originalValue instanceof Date
+          && value.getTime() === originalValue.getTime()))
+    ) {
+      return true;
+    }
+
+    // not changed when set to same empty value
+    if (!originalValue && !value && originalValue === value) {
+      return true;
+    }
+
+    return false;
+  }
+
+  protected _applyTimezone(date: AcceptedDate, options: { timezone?: string | undefined }) {
+    if (options.timezone) {
+      if (isValidTimeZone(options.timezone)) {
+        return dayjs(date).tz(options.timezone);
+      }
+
+      return dayjs(date).utcOffset(options.timezone);
+    }
+
+    return dayjs(date);
+  }
+
+  toBindableValue(
+    date: AcceptedDate,
+    options: StringifyOptions,
+  ) {
+    // Z here means current timezone, _not_ UTC
+    return this._applyTimezone(date, options).format('YYYY-MM-DD HH:mm:ss.SSS Z');
+  }
+}
+
+/**
+ * A date only column (no timestamp)
+ *
+ * Fallback Policy:
+ * If the dialect does not support this type natively, it will be replaced by a string type,
+ * and a CHECK constraint to enforce a valid ISO 8601 datetime format.
+ */
+export class DATEONLY extends AbstractDataType<AcceptedDate> {
+  static readonly [kDataTypeIdentifier]: string = 'DATEONLY';
+
+  toSql() {
+    return 'DATE';
+  }
+
+  toBindableValue(date: AcceptedDate, _options: StringifyOptions) {
+    return dayjs(date).format('YYYY-MM-DD');
+  }
+
+  sanitize(value: unknown): unknown {
+    if (typeof value !== 'string' && typeof value !== 'number' && !(value instanceof Date)) {
+      throw new TypeError(`${value} cannot be normalized into a DateOnly string.`);
+    }
+
+    if (value) {
+      return dayjs(value).format('YYYY-MM-DD');
```

Comment:
possible issue here: https://github.com/sequelize/sequelize/issues/10982
Need to add a test case for this.

Reply:
Feel free to add that test here, or add a TODO for the test

Reply:
It doesn't seem to be an issue with dayjs, I couldn't reproduce it

Reply:
Actually the problem is likely still there, just hidden 
![image](https://user-images.githubusercontent.com/1280915/200176339-a596885d-3107-48e8-b7ec-781d80c3248f.png)

I'll parse it as UTC & add a test

Reply:
Done

Reply:
This change looks good

Reply:
Just need to update the integration test suite to use the new globalTimeZoneConfig; https://github.com/sequelize/sequelize/blob/ephys/ts-data-types/test/integration/timezone.test.js